raiden-ts/src/channels/epics/settle.ts

```diff
@@ -66,56 +67,56 @@ export function channelSettleableEpic(
+ * waiting (earlier or later) channel isn't settling/settled, we do it anyway.
  *
  * @param action$ - Observable of RaidenActions
  * @param state$ - Observable of RaidenStates
  * @param deps - RaidenEpicDeps members
  * @param deps.config$ - Config observable
+ * @param deps.address - Our address
  * @returns Observable of channelSettle.request actions
  */
 export function channelAutoSettleEpic(
-  {}: Observable<RaidenAction>,
+  action$: Observable<RaidenAction>,
   state$: Observable<RaidenState>,
-  { config$ }: RaidenEpicDeps,
+  { address, config$ }: RaidenEpicDeps,
 ): Observable<channelSettle.request> {
   return state$.pipe(
-    groupChannel(),
-    mergeMap((grouped$) =>
-      grouped$.pipe(
-        filter(
-          (channel): channel is Channel & { state: ChannelState.settleable } =>
-            channel.state === ChannelState.settleable,
-        ),
-        take(1),
-        withLatestFrom(config$),
-        // wait [confirmationBlocks, 2 * confirmationBlocks] before proceeding
-        delayWhen(([channel, { confirmationBlocks }]) => {
-          const settleBlock =
-            channel.closeBlock +
-            channel.settleTimeout +
-            Math.round(confirmationBlocks * (1.0 + Math.random()));
-          return state$.pipe(
-            pluck('blockNumber'),
-            filter((blockNumber) => settleBlock <= blockNumber),
-            take(1),
-          );
-        }),
-        withLatestFrom(state$),
-        // filter channel isn't yet being settled by us or partner (state=settling)
-        filter(
-          ([[channel], state]) =>
-            state.channels[channelKey(channel)]?.state === ChannelState.settleable,
-        ),
-        map(([[channel]]) =>
-          channelSettle.request(undefined, {
-            tokenNetwork: channel.tokenNetwork,
-            partner: channel.partner.address,
-          }),
-        ),
-      ),
-    ),
+    distinctUntilKeyChanged('blockNumber'),
+    withLatestFrom(action$.pipe(getPresencesByAddress()), config$),
+    mergeMap(function* ([
+      { blockNumber, channels },
+      presences,
+      { confirmationBlocks, revealTimeout },
+    ]) {
+      for (const channel of Object.values(channels)) {
+        if (channel.state !== ChannelState.settleable) continue;
+
+        const partnerIsOnlineLC =
+          channel.partner.address in presences &&
+          !getCap(presences[channel.partner.address]?.payload.caps, Capabilities.DELIVERY);
+
+        /* iff we are *sure* partner will wait longer (i.e. they're a LC and *we* are the closing
+         * side), then we autoSettle early; otherwise (it's a PC or they're the closing side),
+         * we can wait longer before autoSettling */
+        let waitConfirmations: number;
+        if (partnerIsOnlineLC && channel.closeParticipant === address)
+          waitConfirmations = confirmationBlocks;
+        else waitConfirmations = revealTimeout;
```

Comment:
Shouldn't this be `revealTimeout + confirmationBlocks`?

Reply:
No need.. `revealTimeout` is just a bigger value than `confirmationBlocks`, to shift the delay; no need to _confirm_ it